src/editor/grid.rs

```diff
@@ -147,10 +149,11 @@ impl CharacterGrid {
                     }
                 }
             }
+            false
         }
     }
 
-    fn get_row_array_index(&self, index: isize) -> usize {
+    pub fn get_row_array_index(&self, index: isize) -> usize {
```

Comment:
This makes me think a higher level function which ensures you access the correct line ever time would be wise. Maybe something like get_line and get_line_mut? That way the lines vec can be hidden way

Reply:
Leaving this for Fredizzimo who wrote this originally

Reply:
This is abstracted by a RingBuffer now.